examples/reaction/gray_scott_io.py

```diff
@@ -0,0 +1,89 @@
+"""
+Gray-Scott diffusion-reaction demo, taken from
+[Hundsdorf & Verwer 2003]. It also appears as a
+PETSc TS tutorial and can be found in your Firedrake
+installation at
+
+  $VIRTUAL_ENV/src/petsc/src/ts/tutorials/advection-diffusion-reaction/ex5.c
+
+The test case consists of two tracer species, which
+experience different diffusivities and react with one
+another. The problem is interesting from a numerical
+modelling perspective because of the nonlinear coupling
+between the two equations in the system. Instead of
+solving the equations as a monolithic system, we opt to
+solve them alternately using a Picard iteration.
+
+[Hundsdorf & Vermer 2003] W. Hundsdorf & J.G. Verwer
+    (2003). "Numerical Solution of Time-Dependent
+    Advection-Diffusion-Reaction Equations", Springer.
+"""
+from thetis import *
+
+
+# Doubly periodic domain
+mesh2d = PeriodicSquareMesh(65, 65, 2.5, quadrilateral=True, direction='both')
+x, y = SpatialCoordinate(mesh2d)
+
+# Arbitrary bathymetry
+P1_2d = get_functionspace(mesh2d, "CG", 1)
+bathymetry2d = Function(P1_2d).assign(1.0)
+
+# Setup solver object
+solver_obj = solver2d.FlowSolver2d(mesh2d, bathymetry2d)
+options = solver_obj.options
+options.tracer_only = True
+options.tracer_element_family = 'cg'
+options.use_supg_tracer = False
+options.use_limiter_for_tracers = False
+sim_end_time = 2000.0
+
+# Specify iterative method
+options.tracer_picard_iterations = 2
+options.set_timestepper_type('CrankNicolson', implicitness_theta=1.0)
+
+# Load tracers
+solver_obj.create_function_spaces()
+solver_obj.load_tracers_2d(
+    "gray-scott", input_directory="reaction_models",
```

Comment:
If I run locally, it can't find the yaml file. Maybe move it to `examples/reaction`?

Reply:
Done.